x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/security/authz/permission/FieldPermissionsTests.java

```diff
@@ -82,23 +85,63 @@ public void testFieldPermissionsIntersection() {
         }
     }
 
+    public void testMultipleLimiting() {
+        // Basic test for a number of permission definitions
+        FieldPermissions fieldPermissions = FieldPermissions.DEFAULT;
+        final int nSets = randomIntBetween(2, 8);
+        final FieldPermissionsDefinition fieldPermissionsDefinition = fieldPermissionDef(
+            new String[] { "f1", "f2", "f3*" },
+            new String[] { "f3" }
+        );
+        for (int i = 0; i < nSets; i++) {
+            fieldPermissions = fieldPermissions.limitFieldPermissions(new FieldPermissions(fieldPermissionsDefinition));
+        }
+        final List<FieldPermissionsDefinition> fieldPermissionsDefinitions = fieldPermissions.getFieldPermissionsDefinitions();
+        assertNonNullFieldPermissionDefinitions(fieldPermissionsDefinitions, nSets);
+        fieldPermissionsDefinitions.forEach(fpd -> assertThat(fpd, equalTo(fieldPermissionsDefinition)));
+        assertThat(fieldPermissions.grantsAccessTo(randomFrom("f1", "f2", "f31")), is(true));
+        assertThat(fieldPermissions.grantsAccessTo(randomFrom("f1", "f2", "f3")), is(false));
```

Comment:
This line fails: https://gradle-enterprise.elastic.co/s/7a72jh2nv36zi/tests/overview?outcome=failed

Reply:
Err it's a copy paste error from the above line. It should just test `f3` instead of randomising. Thanks for pointing it out.